comp/netflow/server/server.go

```diff
@@ -123,3 +142,58 @@ func (s *Server) Stop() {
 	}
 	s.running = false
 }
+
+func extractListenerDetails(listeners []*netflowListener) []map[string]interface{} {
+	details := make([]map[string]interface{}, 0, len(listeners))
+
+	for _, listener := range listeners {
+		lDetail := map[string]interface{}{
+			"BindHost":  listener.config.BindHost,
+			"FlowType":  listener.config.FlowType,
+			"Port":      listener.config.Port,
+			"Workers":   listener.config.Workers,
+			"Namespace": listener.config.Namespace,
+		}
+
+		if listener.Error != nil {
+			lDetail["Error"] = listener.Error.Error()
+		} else {
+			lDetail["Error"] = "No Errors"
+		}
+
+		details = append(details, lDetail)
+	}
+	return details
+}
+
+// IsEnabled checks if the netflow functionality is enabled in the configuration.
+func IsEnabled() bool {
+	return config.Datadog.GetBool("network_devices.netflow.enabled")
+}
+
+// GetStatus retrieves the current status of the server with details about
+// all listeners and categorizes them into working and closed.
+func GetStatus() NetflowServerStatus {
+	workingListeners := []*netflowListener{}
+	closedListenersList := []*netflowListener{}
+
+	for _, listener := range globalServer.listeners {
+		if listener.Error != nil {
+			closedListenersList = append(closedListenersList, listener)
+		} else {
+			workingListeners = append(workingListeners, listener)
+		}
+	}
+
+	statusInstance.TotalListeners = int(len(globalServer.listeners))
+	statusInstance.OpenListeners = int(len(workingListeners))
+	statusInstance.ClosedListeners = int(len(closedListenersList))
+
+	return NetflowServerStatus{
+		TotalListeners:         int(statusInstance.TotalListeners),
+		OpenListeners:          int(statusInstance.OpenListeners),
+		ClosedListeners:        int(statusInstance.ClosedListeners),
```

Comment:
Do any of these need to be expvars? I know we use those for status info in the traps server but I don't know if we care about that here.